### ISSUE TITLE
fastlane deliver error with ruby 2.5.0

### ISSUE THREAD

**Issue by rogomantik** [fastlane/fastlane#11395]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
fastlane deliver error

##### Complete output when running fastlane, including the stack trace and command used
run

> fastlane deliver --skip_screenshots

get

> [⠋] 🚀 /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/user_interaction.rb:334: warning: constant ::Data is deprecated

> ...
> Loading './metadata/review_information/notes.txt'...

> Traceback (most recent call last):
> 	18: from /Users/user/.rbenv/versions/2.5.0/bin/fastlane:23:in `<main>'
> 	17: from /Users/user/.rbenv/versions/2.5.0/bin/fastlane:23:in `load'
> 	16: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/bin/fastlane:20:in `<top (required)>'
> 	15: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/cli_tools_distributor.rb:83:in `take_off'
> 	14: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/commands_generator.rb:12:in `start'
> 	13: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/commands_generator.rb:177:in `run'
> 	12: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
> 	11: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
> 	10: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
> 	 9: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
> 	 8: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
> 	 7: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/commands_generator.rb:74:in `block (2 levels) in run'
> 	 6: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:25:in `run'
> 	 5: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:106:in `upload_metadata'
> 	 4: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:162:in `validate_html'
> 	 3: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:10:in `run'
> 	 2: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:44:in `render'
> 	 1: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/2.5.0/erb.rb:885:in `result'
> /Users/user/.rbenv/versions/2.5.0/lib/ruby/2.5.0/erb.rb:885:in `eval': **(erb):265: syntax error, unexpected ')', expecting end-of-input (SyntaxError)
> );  if @options[:trade_represe...
> ^**

### Environment

[⠋] 🚀 /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/user_interaction.rb:334: warning: constant ::Data is deprecated
[✔] 🚀 
[12:12:07]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[12:12:07]: Generating fastlane environment output, this might take a few seconds...
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.2                                     |
| Ruby                        | 2.5.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.rbenv/versions/2.5.0/bin/fastlane        |
| Host                        | Mac OS X 10.13.2 (17C88)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.0/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2n  7 Dec 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | it_IT.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.72.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.2.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.7.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.2       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| bundler                   | 1.16.1       |
| date                      | 1.0.0        |
| fileutils                 | 1.0.2        |
| ipaddr                    | 1.2.0        |
| openssl                   | 2.1.0        |
</details>

*generated on:* **2018-01-03**
</details>

**Comment by KrauseFx** [comment]
Looks like there is a syntax error in your Deliverfile, can you share it with us?

**Comment by rogomantik** [comment]
Here


app_identifier "The.Iden-tifier" # The bundle identifier of your app
username "myemail@icloud.com" # your Apple ID user

copyright "#{Time.now.year} D&veloper."

automatic_release true

**Comment by KrauseFx** [comment]
```
2.72.0/deliver/lib/deliver/runner.rb:162:in validate_html' 3: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:10:inrun'
2: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:44:in render' 1: from /Users/user/.rbenv/versions/2.5.0/lib/ruby/2.5.0/erb.rb:885:inresult'
/Users/user/.rbenv/versions/2.5.0/lib/ruby/2.5.0/erb.rb:885:in `eval': (erb):265: syntax error, unexpected ')', expecting end-of-input (SyntaxError)
); if @options[:trade_represe.
```

mh, weird, something is wrong, maybe because of the `&` in your `Deliverfile`?

**Comment by rogomantik** [comment]
I just tried removing the line that contains & but the problem persists
worked before the ruby update to 2.5.0 using rbenv
It worked with version 2.4.x
fastlane gives the warning at start
/gems/commander-fastlane-4.4.5/lib/commander/user_interaction.rb:334: warning: constant ::Data is deprecated
I also checked /.rbenv/versions/2.5.0/lib/ruby/2.5.0/erb.rb for the syntax error, unexpected ')'
But Sublimetext builds the file correctly

**Comment by rogomantik** [comment]
Just installed ruby 2.4.3 and works

**Comment by casz** [comment]
Got very similar result:

```
[11:04:09]: fastlane finished with errors
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
SyntaxError: (erb):265: syntax error, unexpected ')', expecting end-of-input
);  if @options[:trade_represe...
^
  /usr/local/Cellar/ruby/2.5.0/lib/ruby/2.5.0/erb.rb:885:in `eval'
  /usr/local/Cellar/ruby/2.5.0/lib/ruby/2.5.0/erb.rb:885:in `result'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:44:in `render'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/html_generator.rb:10:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:162:in `validate_html'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:106:in `upload_metadata'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/deliver/lib/deliver/runner.rb:25:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:16:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
  Fastfile:206:in `block (2 levels) in parsing_binding'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/command_line_handler.rb:34:in `handle'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:343:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/cli_tools_distributor.rb:91:in `take_off'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.72.0/bin/fastlane:20:in `<top (required)>'
  /usr/local/bin/fastlane:23:in `load'
  /usr/local/bin/fastlane:23:in `<top (required)>'
VASOSXDEV02:cfdapp appbuilder$ cat fastlane/Deliverfile 
copyright "© #{Time.now.year} Widex A/S"
app_rating_config_path './fastlane/rating_config.json'
app_icon './App.iOS/Resources/Images.xcassets/AppIcons.appiconset/iconLanuch_1024.png'
```

**Comment by lacostej** [comment]
The `if @options[:trade_represe` comes from deliver/lib/assets/summary.html.erb

The preceding `);` is suspicious. Have we run automated tests under ruby 2.5.0? It should excercise some of it.

**Comment by lacostej** [comment]
I installed ruby 2.5.0 

````
rspec ./deliver/spec/html_generator_spec.rb:18 # Deliver::HtmlGenerator render minimal configuration renders HTML
rspec ./deliver/spec/html_generator_spec.rb:32 # Deliver::HtmlGenerator render with keywords renders HTML
````

definitively a ruby 2.5.0 incompatibility

**Comment by welshm** [comment]
Was this solved? I'm still seeing this warning on 2.81.0

slave1@slave2:~$ fastlane -v
fastlane installation at path:
/usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.81.0/bin/fastlane
-----------------------------
[⠙] 🚀 /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.5/lib/commander/user_interaction.rb:334: warning: constant ::Data is deprecated
[✔] 🚀 
fastlane 2.81.0
